src/node/plugin-mdx/pluginMdxGit.ts

```diff
@@ -0,0 +1,36 @@
+import { Plugin } from 'vite';
+import { simpleGit } from 'simple-git';
+import { MD_REGEX } from '../../node/constants';
+
+export function pluginMdxGit(): Plugin {
+  const cache = new Map<string, string>();
+  const git = simpleGit();
+
+  // https://github.com/steveukx/git-js#git-log
+  async function getLastUpdatedTime(path: string) {
+    const { latest } = await git.log({ file: path });
+    return !latest ? '' : latest.date;
+  }
+
+  return <Plugin>{
+    name: 'vite-plugin-mdx-git',
+    async transform(code, id) {
+      if (!MD_REGEX.test(id)) return code;
+
+      let lastUpdatedTime = '';
+      if (cache.has(id)) {
+        lastUpdatedTime = id;
```

Comment:
Nice! `getLastUpdateTime` will use git command, which means expensive cost, the cache will avoid the cost after hmr. But it's better to enable updating timestamp after hmr, as the optional config, such as `siteConfig.lastUpdated`.When user set it to `true`, we cancel the cache and get the latest timestamp in the page after hmr, with the more corrective info as well as the more performance cost.

Reply:
About hmr implement, you can refer #43

Reply:
> About hmr implement, you can refer #43

https://github.com/sanyuan0704/island.js/pull/43.This hmr processing scheme has bug.When modifying the md file ,the last update timestamp is not persistent, and the update time will be lost after switching routes

Reply:
https://github.com/sanyuan0704/island.js/pull/43

Now, I'm dealing this with hmr, but I think my current practice is wrong

Reply:
@lweijian Oh, i seem have some mistake because `getLastUpdateTime` get the timestamp from git history, so there is no need to handle hmr with it. Yeah, i think this pr can work directly.

Reply:
Wait, I think I might made something wrong. `transform` is a sequential hook and will only be called once during building process. So we may don't need `cache` here?